cfme/common/provider.py

```diff
@@ -35,15 +89,411 @@ def get_yaml_data(self):
             raise ProviderHasNoKey('Provider %s has no key, so cannot get yaml data', self.name)
 
     def get_mgmt_system(self):
-        """ Returns the mgmt_system using the :py:func:`utils.providers.provider_factory` method.
+        """ Returns the mgmt_system using the :py:func:`utils.providers.get_mgmt` method.
         """
         if hasattr(self, 'provider_data') and self.provider_data is not None:
-            return provider_factory(self.provider_data)
+            return get_mgmt(self.provider_data)
         elif self.key is not None:
-            return provider_factory(self.key)
+            return get_mgmt(self.key)
         else:
             raise ProviderHasNoKey('Provider %s has no key, so cannot get mgmt system')
 
+    def _submit(self, cancel, submit_button):
+        if cancel:
+            form_buttons.cancel()
+            # sel.wait_for_element(page.configuration_btn)
+        else:
+            submit_button()
+            flash.assert_no_errors()
+
+    def create(self, cancel=False, validate_credentials=False):
+        """
+        Creates a provider in the UI
+
+        Args:
+           cancel (boolean): Whether to cancel out of the creation.  The cancel is done
+               after all the information present in the Provider has been filled in the UI.
+           validate_credentials (boolean): Whether to validate credentials - if True and the
+               credentials are invalid, an error will be raised.
+        """
+        sel.force_navigate('{}_provider_new'.format(self.page_name))
+        fill(self.properties_form, self._form_mapping(True, **self.__dict__))
+        for cred in self.credentials:
```

Comment:
Credentials are now special in that they are a dict of Credential objects. Potential things I did not think of yet, making sure that each credential filler has the right "tab" to click on.